tests/unit/storage/test_migrate.py

```
@@ -77,12 +77,19 @@ def test_migration_failure_raises_runtime_error(self) -> None:
+        alembic.ini in a non-source-checkout install) must fail loudly here
+        rather than defer to a confusing "relation does not exist" error
+        on the first query.
+        """
         from cloud_agents.storage.migrate import run_alembic
 
         with patch("cloud_agents.storage.migrate._ALEMBIC_INI") as mock_ini:
             mock_ini.exists.return_value = False
 
-            # Should not raise
-            run_alembic("postgresql://localhost/testdb")
+            with pytest.raises(RuntimeError, match="alembic.ini not found"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)